### PR TITLE
feat: Sidepanel overflow menu

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

We are using the side panel more and more. It could make sense to actually have a menu to get to all options that _could_ open there, even though they are not typically shown.

Longer term we could turn this into a checkilist so you could actually choose which items are shown in the side panel giving it a personal touch.

## Changes

* Hides the activity tab by default
* Refactors the sidebar layout for better overflow support
* Adds a dropdown menu for "secret" panels

Screenshot 2023-11-29 at 10 27 08

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/layout/navigation-3000/sidepanel/SidePanel.scss [hunk 1]
```diff
@@ -36,15 +38,31 @@
...
+        .SidePanel3000__bar__tabs {
+            flex: 1;
+            width: var(--side-panel-bar-width);
+            overflow: hidden auto;
+
+            &::-webkit-scrollbar {
+                display: none;
+            }
+
+            .SidePanel3000__bar__tabs__rotate {
```
Comment (Twixes) [inline frontend/src/layout/navigation-3000/sidepanel/SidePanel.scss:57]: These classnames go against [BEM](https://getbem.com/) concept – `SidePanel3000__bar__tabs__rotate` isn't Block Element, it's Block Element Element Element. So we still end up with a deeply nested structure AND monstrous class names.

`SidePanel3000__bar__tabs` should be just something like `SidePanel3000__tabs`, and `SidePanel3000__bar__tabs__rotate` - `SidePanel3000__tabs-content`. Both should also be at the top level of the file, there's no point nesting them (uncontrollable nesting is THE thing that BEM solves).

File: frontend/src/layout/navigation-3000/sidepanel/sidePanelLogic.tsx [hunk 2]
```diff
@@ -82,5 +78,12 @@ export const sidePanelLogic = kea<sidePanelLogicType>([
                 })
             },
         ],
+
+        secretTabs: [
```
Comment (Twixes) [inline frontend/src/layout/navigation-3000/sidepanel/sidePanelLogic.tsx:82]: Not so secret anymore, but maybe `extraTabs` or `ephemeralTabs`?

**Review by Twixes — CHANGES_REQUESTED** [review]
Love this solution.
Just got some reservations about the CSS. 😅

File: frontend/src/layout/navigation-3000/sidepanel/SidePanel.scss (hunk 1 above)
Comment (benjackwhite) [reply frontend/src/layout/navigation-3000/sidepanel/SidePanel.scss:57]: Hmmmm I guess so. I like the alternative interpretation of BEM but sure happy to change it

**Pull request merged**